docs/source/tutorials/fashion_mnist.ipynb

```diff
@@ -0,0 +1,1257 @@
+    "\n",
+    "            # zero the parameter gradients\n",
+    "            optimizer.zero_grad()\n",
+    "\n",
+    "            # forward + backward + optimize\n",
+    "            outputs = model(inputs)\n",
+    "            loss = criterion(outputs, labels)\n",
+    "            loss.backward()\n",
+    "            optimizer.step()\n",
+    "\n",
+    "            # print statistics\n",
+    "            running_loss += loss.item()\n",
+    "\n",
+    "        accuracy = get_test_accuracy(model, testloader)\n",
+    "        print(\n",
+    "            f\"epoch: {epoch + 1} loss: {running_loss / len(trainloader):.3f} test acc: {accuracy}\"\n",
+    "        )\n",
+    "\n",
+    "        running_loss = 0.0\n",
+    "\n",
+    "        if accuracy > best_test_accuracy:\n",
+    "            best_epoch = epoch\n",
+    "\n",
+    "        if epoch - best_epoch > patience:\n",
+    "            print(f\"Early stopping at epoch {epoch + 1}\")\n",
+    "            break\n",
+    "    return model\n",
+    "\n",
+    "\n",
+    "# Method for computing out-of-sample embeddings\n",
+    "def compute_embeddings(model, testloader):\n",
+    "    embeddings_list = []\n",
+    "\n",
+    "    with torch.no_grad():\n",
+    "        for data in tqdm(testloader):\n",
+    "            images, labels = data[\"image\"].to(device), data[\"label\"].to(device)\n",
+    "\n",
+    "            embeddings = model.embeddings(images)\n",
+    "            embeddings_list.append(embeddings.cpu())\n",
+    "\n",
+    "    return torch.vstack(embeddings_list)\n",
+    "\n",
+    "\n",
+    "# Method for computing out-of-sample predicted probabilities\n",
+    "def compute_pred_probs(model, testloader):\n",
+    "    pred_probs_list = []\n",
+    "\n",
+    "    with torch.no_grad():\n",
+    "        for data in tqdm(testloader):\n",
+    "            images, labels = data[\"image\"].to(device), data[\"label\"].to(device)\n",
+    "\n",
+    "            outputs = model(images)\n",
+    "            pred_probs_list.append(outputs.cpu())\n",
+    "\n",
+    "    return torch.vstack(pred_probs_list)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Create K folds of the dataset for cross validation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "K = 3  # Set to small value here to ensure quick runtimes, we recommend 5 or 10 in practice for more accurate estimates.\n",
+    "n_epochs = 3  # Number of epochs for to train the model on. This is set to a small value to keep the runtime value small, but it is typically more for more complex images and models.\n",
+    "patience = 2  # Parameter for early stopping. If the validation accuracy does not improve for this many epochs, training will stop.\n",
+    "batch_size = 4  # Batch size for training and testing\n",
```

Comment:
why did you opt for such a tiny batch size?

Reply:
The results get worsened using a higher batch size